### PR TITLE
Upgrade to Rails 5.2

### LINKED ISSUE #600: Upgrade from Rails 4.2.8 to Rails 5.0.1
Follow the steps below will definitely help:
https://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#upgrading-from-rails-4-2-to-rails-5-0
This depends on #481 and #594

### PR DESCRIPTION
**Pull request by kaustubh-nair**
Fixes #600

### CODE REVIEW

**Comment by jywarren** [comment]
exciting!!!!!!

File: Gemfile.lock [hunk 1]
```diff
@@ -297,7 +297,7 @@ DEPENDENCIES
   passenger
   popper_js (~> 1.11, >= 1.11.1)
   pry-rails (~> 0.3.9)
-  rails (~> 5.0)
+  rails (~> 5.2.3)
   rails-perftest
```
Comment (kaustubh-nair) [inline Gemfile.lock]: We're going straight to 5.2? @alaxalves ?

Comment (alaxalves) [reply Gemfile.lock]: Yes, it was a nice suggestion of @IgorWilbert. Since the code is small I think it's a good option we jump straight to 5.2.3

**Comment by kaustubh-nair** [comment]
@jywarren @sashadev-sky We are stuck at trying to get the database setup. Can you please help us out here?
Thanks

File: app/models/annotation.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
-class Annotation < ActiveRecord::Base
-  include ActiveModel::MassAssignmentSecurity
-  belongs_to :map
-  belongs_to :user
+class Annotation < ApplicationRecord
+  
+  # FIXME - We should start using strong parameters here in favor of this below  
+  # include ActiveModel::MassAssignmentSecurity
+  belongs_to :map, optional: true
+  belongs_to :user,optional: true
```
Comment (kaustubh-nair) [inline app/models/annotation.rb]: Why are the attributes optional?

Comment (alaxalves) [reply app/models/annotation.rb]: This a new configuration of Rails >5

**Comment by ViditChitkara** [comment]
Hi, are we going to merge this soon?? I think we can setup action cable (which requires rails 5) to carry out the synchronous image management part. @jywarren @kaustubh-nair @alaxalves .
Here are the issue details #609

**Comment by ViditChitkara** [comment]
Should I make a branch over it and setup the action cable part here. (I think it might be stable enough to get the action cable work going).
Thanks!

**Comment by kaustubh-nair** [comment]
> Hi, are we going to merge this soon?? I think we can setup action cable (which requires rails 5) to carry out the synchronous image management part. @jywarren @kaustubh-nair @alaxalves @jainaman224 .
> Here are the issue details #609

Sure @ViditChitkara, that sounds awesome
We are currently going to deploy Rails 4 to the live site, and after a few days of testing and bug fixing we might merge rails 5 as well.
You can make a new pull to the Rails 5 branch so it becomes easier to review. We can merge it here on this pull when it's ready
Thanks!

**Comment by ViditChitkara** [comment]
That's great! Will do That.

**Comment by alaxalves** [comment]
> That's great! Will do That.

We'll soon merge this to `development` branch, then later you could open your Action Cable setup to `development` branch. We have adopted the pipeline of opening PRs to `development` then `development` to `unstable` , then `unstable` to `main`.

**Comment by kaustubh-nair** [comment]
@alaxalves We'll have to reset the rails-4 commits on `development` because they were squashed and merged on main. Also there will be a lot of conflicts when trying to merge `development` to `main`

**Comment by alaxalves** [comment]
> @alaxalves We'll have to reset the rails-4 commits on `development` because they were squashed and merged on main. Also there will be a lot of conflicts when trying to merge `development` to `main`

We when squash and merge we lose the commits reference, we have to get those conflicts solved, I can do that. :D

**Comment by ViditChitkara** [comment]
> We'll soon merge this to development branch, then later you could open your Action Cable setup to development branch. We have adopted the pipeline of opening PRs to development then development to unstable , then unstable to main.

Sure, I'll start over the development branch!

**Pull request merged**